File: blaze_flutter_ima_ads/ios/Sources/BlazeFlutterIMAPlugin.swift
import Flutter
import UIKit

public class BlazeFlutterIMAPlugin: NSObject, FlutterPlugin {
    
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(name: "blaze_flutter_ima", binaryMessenger: registrar.messenger())
        let instance = BlazeFlutterIMAPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
        
        let messenger = registrar.messenger()
        
        BlazeFlutterIMAModule.registerModule(messenger: messenger)
    }
    
    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "getPlatformVersion":
            result("iOS " + UIDevice.current.systemVersion)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
    
}

File: blaze_flutter_ima_ads/android/src/main/kotlin/com/blaze/flutterblazeima/BlazeFlutterIMAPlugin.kt
package com.blaze.flutterblazeima

import android.app.Application
import androidx.annotation.Keep
import com.blaze.flutterblazeima.sdk_module.BlazeFlutterIMAModule
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** BlazeFlutterIMAPlugin */
@Keep
class BlazeFlutterIMAPlugin : FlutterPlugin, MethodCallHandler {
    /// The MethodChannel that will the communication between Flutter and native Android
    ///
    /// This local reference serves to register the plugin with the Flutter Engine and unregister it
    /// when the Flutter Engine is detached from the Activity
    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "blaze_flutter_ima")
        channel.setMethodCallHandler(this)

        BlazeFlutterIMAModule.registerModule(
            messenger = flutterPluginBinding.binaryMessenger,
            application = (flutterPluginBinding.applicationContext as Application)
        )

    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        if (call.method == "getPlatformVersion") {
            result.success("Android ${android.os.Build.VERSION.RELEASE}")
        } else {
            result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}

